src/components/KAlert/KAlert.vue

```diff
@@ -1,578 +1,185 @@
-      }"
+    <div
+      v-if="!hideIcon"
+      class="alert-icon-container"
     >
       <slot name="icon">
-        <KIcon
-          class="k-alert-icon"
-          :color="iconColor"
-          :icon="icon"
-          :size="iconSize"
+        <component
+          :is="getAlertIcon"
+          class="alert-icon"
+          :size="KUI_ICON_SIZE_40"
         />
       </slot>
-    </span>
-    <!-- Main content of alert -->
-    <div
-      :class="[
-        'k-alert-msg-text',
-        { 'has-dismiss-icon': dismissType === 'icon' },
-        { 'has-dismiss-button': dismissType === 'button' }
-      ]"
-    >
-      <div
-        v-if="title || $slots.title"
-        class="k-alert-title"
+    </div>
+    <div class="alert-content">
+      <span
+        v-if="title"
+        class="alert-title"
       >
-        <slot name="title">
-          {{ title }}
-        </slot>
-      </div>
+        {{ title }}
+      </span>
       <div
-        class="k-alert-msg"
-        :class="{
-          'k-alert-text': size === 'large',
-          'k-alert-subtext': title || $slots.title
-        }"
+        v-if="message || $slots.default"
+        class="alert-message"
       >
-        <!-- @slot Use this slot to pass default alert message  -->
-        <slot name="alertMessage">
-          {{ alertMessage }}
-        </slot>
-      </div>
-
-      <div
-        v-if="size === 'large' && (description || $slots.description)"
-        class="k-alert-description-text"
-      >
-        <!-- @slot Use this slot to pass alert message description for large banner type alert  -->
-        <slot name="description">
-          {{ description }}
+        <slot>
+          <p>{{ message }}</p>
         </slot>
       </div>
     </div>
-    <!-- Buttons -->
-    <button
-      v-if="dismissType === 'icon'"
-      aria-label="Close"
-      class="close"
-      type="button"
-      @click="dismissAlert"
-    >
-      <KIcon
-        :class="appearance"
-        :color="appearance"
-        icon="close"
-        :size="KUI_ICON_SIZE_20"
-      />
-    </button>
-
-    <div
-      v-if="hasActionButtons || dismissType !== 'none'"
-      class="k-alert-action"
-      :class="appearance"
-    >
-      <!-- @slot Use this slot to pass extra buttons other than Dismiss  -->
-      <slot
-        v-if="hasActionButtons"
-        name="actionButtons"
-      >
-        <KButton
-          size="small"
-          @click="proceed"
-          @keyup.enter="proceed"
-        />
-      </slot>
-
-      <KButton
-        v-if="dismissType === 'button'"
-        size="small"
-        @click="dismissAlert"
-      >
-        Dismiss
-      </KButton>
-    </div>
+    <CloseIcon
+      v-if="dismissible"
+      class="alert-dismiss-icon"
+      :size="KUI_ICON_SIZE_40"
+      tabindex="0"
+      @click="$emit('dismiss')"
+      @keydown.space.prevent
+      @keyup.enter="$emit('dismiss')"
+      @keyup.space="$emit('dismiss')"
```

Comment:
I think I've asked this before, but why are you using keyup instead of keydown? Keydown would be the norm.